### PR TITLE
approximate sentence match logic adapter and its docs 

### PR DESCRIPTION
**Pull request by sumn2u**
Jaccard index is found to have better performance in  approximate sentence matching so implement it.

### CODE REVIEW

File: chatterbot/adapters/logic/approximate_sentence_match.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                # Use a randomly picked statement
+                return 0, self.context.storage.get_random()
+            else:
+                raise self.EmptyDatasetException()
+
+        confidence = -1
+        sentence_match = input_statement
+        # Find the  matching known statement
+        for statement in statement_list:
+            ratio = self.is_ci_lemma_stopword_set_match(input_statement.text, statement.text)
```
Comment (gunthercox) [inline chatterbot/adapters/logic/approximate_sentence_match.py]: I'm not sure if this works. Where is `ratio` being used?

**Comment by sumn2u** [comment]
ratio is added here if there is ratio it will return result :)

File: chatterbot/adapters/logic/approximate_sentence_match.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            return (pos_tag[0], wordnet.NOUN)
+        elif pos_tag[1].startswith('R'):
+            return (pos_tag[0], wordnet.ADV)
+        else:
+            return (pos_tag[0], wordnet.NOUN)
+
+
+    def is_ci_lemma_stopword_set_match(self,a, b, threshold=0.5):
+        """Check if a and b are matches."""
+        print("ask",a)
```
Comment (gunthercox) [inline chatterbot/adapters/logic/approximate_sentence_match.py]: This print statement will have to be removed before this gets merged.

File: chatterbot/adapters/logic/approximate_sentence_match.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        confidence = -1
+        sentence_match = input_statement
+        # Find the  matching known statement
+        for statement in statement_list:
+            ratio = self.is_ci_lemma_stopword_set_match(input_statement.text, statement.text)
+            if ratio:
+                closest_match = statement
+            else:
+                closest_match = statement
+        return 50, closest_match
```
Comment (gunthercox) [inline chatterbot/adapters/logic/approximate_sentence_match.py]: Should this be returning `ratio` instead of a fixed value of 50?

Comment (sumn2u) [inline chatterbot/adapters/logic/approximate_sentence_match.py]: Ratio is a Boolean value  since our threshold value is 0.5 which is 50 we make 50 :)

Comment (gunthercox) [inline chatterbot/adapters/logic/approximate_sentence_match.py]: 0.5 would actually be correct here. It appears I never documented it anywhere but confidence values are always between 0 and 1.

Comment (sumn2u) [inline chatterbot/adapters/logic/approximate_sentence_match.py]: Yup so I put 0.5 over there :)

**Comment by gunthercox** [comment]
Thank you! Looks good :smile:

**Pull request merged**